src/acceptor.rs

```diff
@@ -173,6 +183,7 @@ impl rustls_acceptor {
     pub extern "C" fn rustls_acceptor_accept(
         acceptor: *mut rustls_acceptor,
         out_accepted: *mut *mut rustls_accepted,
+        out_alert: *mut *mut rustls_accepted_alert,
```

Comment:
This needs a NULL check. Alternately we could treat NULL as a signal for "I'm not interested in sending alerts, don't do anything with this parameter" (and put that in the docs).

Same for rustls_accepted_into_connection below.

Reply:
I added `NULL` checks for the `out_alert` parameters and folded that into the commit that introduced the params.

Afterwards I did a scan of all the other `set_boxed_mut_ptr` callsites and most were missing the same check for their `dst` param. I think prioritizing one of the systematic solutions from https://github.com/rustls/rustls-ffi/issues/380 would be appealing.

> Another possibility would be to take &mut *mut T instead of *mut *mut T, and rely on our existing try_ macros to produce the reference, which we would then know is non-NULL.

I briefly experimented with this but wasn't able to get it working quickly. I'll come back to it, but for the meantime to unblock this landing I added manual null checks throughout in a separate commit: https://github.com/rustls/rustls-ffi/pull/389/commits/7cc74e47c754fa45b62a07601a463e54c7163b00

Reply:
> Alternately we could treat NULL as a signal for "I'm not interested in sending alerts, don't do anything with this parameter"

I'm lightly in favour of making the parameter mandatory to signal that handling the alerts is the recommended course of action so I went with the NULL check instead of this approach.